rfc/0075-faster-failover-and-configuration-push.md

```diff
@@ -0,0 +1,364 @@
+
+[RFC-0048][rfc-0048] describes bootstrap process for the SDK and the KV connections in particular. With Faster Failover
+mechanism implemented, the SDK should not start polling for the nodes, where the KV engine has acknowledged
+`ClustermapChangeNotificationBrief` feature. Such connections are expected to be notified by the KV engine when the
+configuration will be received.t
+
+### Enhancements in Handling the `NotMyVbucket` Status
+
+`DedupeNotMyVbucketClustermap` feature allows to save traffic by not sending configuration, if SDK already seen the same
+revision. 
+
+Several modifications are required in the SDK:
+1. Response handler should tolerate empty response with `NotMyVbucket` status, as the KV engine assumes that the SDK
+   already seen configuration, and there is no newer configuration available. In this case SDK should just retry the
+   operation.
+
+2. If the reponse payload contains body, it contains current configuration, which should be sent to configuration
+   monitor (manager). The SDK should either synchronously apply configuration, create waiting queue for given
+   `epoch`/`revision` pair.
+   Once configuration applied, the SDK must check if new configuration routes the operation to new endpoint or new
+   vbucket on the old endpoint, and *immediately* dispatch operation to new endpoint (or same endpoint in case vbucketID
+   has changed). In any other case, the SDK should send operation to retry orchestrator.
+   ```mermaid
+   flowchart 
+       A(NotMyVbucket) --> B{Empty Body?}
+       B -->|No|C(Apply Configuration)
+       C --> D{Route Operation}
+       D -->|Endpoint Changed| E[Dispatch To<br>New Endpoint]
+       D -->|VBucketID Changed| F[Update VBucketID]
+       F --> G[Dispatch To<br>Same Endpoint]
+       D -->|Everything Else| H[Send To Retry<br>Orchestrator]
+       B -->|Yes|H[Send To Retry<br>Orchestrator]
+   ```
+
+
+Below is a diagram that illustrates an example of the SDK workflow
+
+```mermaid
+sequenceDiagram
+    autonumber
+
+    participant conn_1
+    participant kv_node_1
+    participant retry_orchestrator
+    participant config_monitor
+
+            conn_1    ->>+  kv_node_1:       get("foo", vb=115)
+            kv_node_1 ->>-  conn_1:          NotMyVbucket(config={epoch=1, rev=11, ...})
+
+            conn_1   -->>+  config_monitor: propose confg={epoch=1, rev=11}
+
+
+    critical Check if config route operation to different node of vbucket
+
+        option "foo" still maps to kv_node_1
+            conn_1  -->>+  retry_orchestrator:  retry(get, "foo", reason=NotMyVbucket, epoch=1, rev=11)
+
+        option "foo" does not map to kv_node_1, or vbucket has changed 
+            conn_1    -->+  kv_node_1:       get("foo", vb=new_vbucket)
+            kv_node_1 ->>-  conn_1:          Success()
+    end
+```
+
+# Language Specifics
+
+## Feature Checklist
+
+1. `GetClusterConfigWithKnownVersion` (`0x1d`). The SDK should always supply current configuration version if the
```

Comment:
In the case that we're using known versions and cccp polling, if the node we've polled returns success but no body should we consider that a successful poll and retry again after the poll period or should we try the next node to see if it has a newer config?